cpp/src/arrow/util/bit-util-test.cc

```diff
@@ -41,6 +44,59 @@ static void EnsureCpuInfoInitialized() {
   }
 }
 
+void WriteVectorToWriter(internal::BitmapWriter& writer, const std::vector<int> values) {
+  for (const auto& value : values) {
+    if (value) {
+      writer.Set();
+    } else {
+      writer.Clear();
+    }
+    writer.Next();
+  }
+  writer.Finish();
+}
+
+void BitmapFromVector(const std::vector<int>& values, int64_t bit_offset,
+                      std::shared_ptr<Buffer>* out_buffer, int64_t* out_length) {
+  const int64_t length = values.size();
+  *out_length = length;
+  ASSERT_OK(GetEmptyBitmap(default_memory_pool(), length + bit_offset, out_buffer));
+  auto writer = internal::BitmapWriter((*out_buffer)->mutable_data(), bit_offset, length);
+  WriteVectorToWriter(writer, values);
+}
+
+#define ASSERT_READER_SET(reader)    \
+  do {                               \
+    ASSERT_TRUE(reader.IsSet());     \
+    ASSERT_FALSE(reader.IsNotSet()); \
+    reader.Next();                   \
+  } while (0)
+
+#define ASSERT_READER_NOT_SET(reader) \
+  do {                                \
+    ASSERT_FALSE(reader.IsSet());     \
+    ASSERT_TRUE(reader.IsNotSet());   \
+    reader.Next();                    \
+  } while (0)
+
+// Assert that a BitmapReader yields the given bit values
+void ASSERT_READER_VALUES(internal::BitmapReader& reader, std::vector<int> values) {
+  for (const auto& value : values) {
+    if (value) {
+      ASSERT_READER_SET(reader);
+    } else {
+      ASSERT_READER_NOT_SET(reader);
+    }
+  }
+}
+
+// Assert equal contents of a memory area and a vector of bytes
+void ASSERT_BYTES_EQ(const uint8_t* left, const std::vector<uint8_t>& right) {
+  // No direct conversion from (T*, size) to vector<T>
```

Comment:
Can't you construct it like this?

```c++
std::vector<T> left_array(left, left + right.size());
```
Or am I missing something

Reply:
Hmm... you may be right, I'll give it a try.